client/src/pages/ResourcePage/ResourcePage.js

```diff
@@ -0,0 +1,301 @@
+    }
+
+    toggleCollapseAllFields() {
+        const { fields, collapseAll } = this.state;
+        for (let fieldName in fields) {
+            if (!fields.hasOwnProperty(fieldName)) continue;
+            fields[fieldName].collapsed = !collapseAll;
+        }
+        this.setState({
+            fields,
+            collapseAll: !collapseAll
+        });
+    }
+
+    render() {
+        const { collapseAll, search } = this.state;
+        return (
+            <div className="page-container">
+                <div className="resources page static">
+                    <div className="static-page-body">
+                        <div className="row">
+                            <div className="col">
+                                <div className="static-page-title">
+                                    <h1>Resources and Glossary</h1>
+                                </div>
+                                <div className="static-page-text">
+                                    <p>
+                                        This page provides descriptions, formats, and examples of our system's fields for monuments and memorials, to assist you in submitting changes to our database.
+                                    </p>
+                                    <div className="collapse-all-search-container">
+                                        <div className="d-flex">
+                                            <TextSearch disableAnimation
+                                                        placeholder="Search fields..."
+                                                        value={search}
+                                                        className="form-control"
+                                                        onSearchChange={search => this.setState({search})}
+                                                        onClear={() => this.setState({search: ''})}/>
+                                            <Button variant="light" className="btn-sm collapse-btn"
+                                                    onClick={() => this.toggleCollapseAllFields()}>
+                                                {collapseAll ? 'Show All' : 'Collapse All'}
+                                            </Button>
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+                        </div>
+                        {this.renderFields()}
+                    </div>
+                </div>
+                <Footer/>
+            </div>
+        );
+    }
+
+    renderFields() {
+        const { search } = this.state;
+
+        const fields = [];
+        for (let fieldName in this.state.fields) {
+            if (!this.state.fields.hasOwnProperty(fieldName)) continue;
+            const field = this.state.fields[fieldName];
+            field.content = this.fieldContent[fieldName];
```

Comment:
Can you explain this to me? The syntax for passing a "parameter" (I guess) to a `get` is `[]`?

Reply:
No - `get` functions in ES6 are accessed just like a property - so `this.fieldContent` is like calling `this.filedContent()` if it were a regular function. Since `get fieldContent` returns an object, `fieldContent[fieldName]` returns the value of the property named `fieldName` in the object

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/set

Reply:
Ah I see. This is pog.